pkg/bgpv1/manager/reconcilerv2/paths.go

```diff
@@ -152,34 +202,74 @@ func reconcilePaths(params *reconcilePathsParams) (PathMap, error) {
 
 	// withdraw unneeded adverts
 	for advrtKey, advrt := range toWithdraw {
-		l.WithFields(logrus.Fields{
-			types.PathLogField:   advrt.NLRI.String(),
-			types.FamilyLogField: advrt.Family.String(),
-		}).Debug("Withdrawing path")
-
-		if err := params.Router.WithdrawPath(params.Ctx, types.PathRequest{Path: advrt}); err != nil {
+		err := withdrawPath(params, advrtKey, advrt)
+		if err != nil {
 			return runningAdverts, err
 		}
 		delete(runningAdverts, advrtKey)
 	}
 
 	// create new adverts
 	for advrtKey, advrt := range toAdvertise {
-		l.WithFields(logrus.Fields{
-			types.PathLogField:   advrt.NLRI.String(),
-			types.FamilyLogField: advrt.Family.String(),
-		}).Debug("Advertising path")
-
-		advrtResp, err := params.Router.AdvertisePath(params.Ctx, types.PathRequest{Path: advrt})
+		path, err := advertisePath(params, advrtKey, advrt)
 		if err != nil {
 			return runningAdverts, err
 		}
-		runningAdverts[advrtKey] = advrtResp.Path
+		runningAdverts[advrtKey] = path
 	}
 
 	return runningAdverts, nil
 }
 
+func advertisePath(params *reconcilePathsParams, pathKey string, path *types.Path) (*types.Path, error) {
+	if params.PathReferences != nil && params.PathReferences[pathKey] > 0 {
+		// path already advertised for another resource
+		params.PathReferences[pathKey] += 1
+		return path, nil
```

Comment:
I just realized that returning the `path` (input parameter) here will not work in some cases. We need to return the actual already advertised path, as only that one contains the UUID required for path withdrawal. Will fix this.

Reply:
Ah I missed it too, I think UUID is used when withdrawing a route. Did we had any test which caught it? 

Maybe we need to save the pointer to returned path in PathReferences.

Reply:
>  Did we had any test which caught it?

nope, unfortunately the unit test is not catching this (it is using fake router), I noticed that when working on BGPv1 version of this fix

> Maybe we need to save the pointer to returned path in PathReferences.

yeah, I did that, please review `paths.go` again.